src/Microsoft.OData.Core/Json/ODataJsonResourceDeserializer.cs

```diff
@@ -2989,7 +3060,24 @@ internal async Task ReadODataOrCustomInstanceAnnotationValueAsync(
             {
                 // #### annotation 1 ####
                 // built-in "odata." annotation value is added to propertyAndAnnotationCollector then later to resourceSet.InstanceAnnotations.
-                propertyAndAnnotationCollector.AddODataScopeAnnotation(annotationName, await this.JsonReader.GetValueAsync().ConfigureAwait(false));
+                Task<object> annotationTask = this.JsonReader.GetValueAsync();
+
+                if (annotationTask.IsCompletedSuccessfully)
+                {
+                    object annotationValue = annotationTask.Result;
+                    try
+                    {
+                        propertyAndAnnotationCollector.AddODataScopeAnnotation(annotationName, annotationValue);
+                    }
+                    catch (ODataException ex)
+                    {
+                        return Task.FromException<object>(ex);

```

Comment:
To be consistency with other parts of code, this should likely be `Task.FromException(ex)` without the generic parameter.

```suggestion
                        return Task.FromException(ex);
```